### PR TITLE
eh/filters: send rpctransactions in pending-subscription

### PR DESCRIPTION
**Pull request by diwu1989**
Porting https://github.com/ethereum/go-ethereum/pull/26126 to Celo so that the pending subscription TX has consistent JSON.

This was fixed upstream by always using the `ethapi.NewRPCPendingTransaction` function

### CODE REVIEW

File: eth/filters/api.go [hunk 1]
```diff
@@ -154,15 +154,18 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context, fullTx *
...
 		for {
 			select {
 			case txs := <-txs:
 				// To keep the original behaviour, send a single tx hash in one notification.
 				// TODO(rjl493456442) Send a batch of tx hashes in one notification
+				latest := api.backend.CurrentHeader()
 				for _, tx := range txs {
 					if fullTx != nil && *fullTx {
-						notifier.Notify(rpcSub.ID, tx)
+						rpcTx := ethapi.NewRPCPendingTransaction(tx, latest, chainConfig, nil)
```
Comment (palango) [inline eth/filters/api.go:167]: Could you add a comment here mentioning that a `basefeefn` is not required here, because it's a pending transaction?
See https://github.com/celo-org/celo-blockchain/blob/f0adb14e48288ff6d6e4d9320d5980afd4d17c2d/internal/ethapi/api.go#L1340-L1354

File: eth/filters/filter_system.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ const (
 	PendingTransactionsSubscription
 	// BlocksSubscription queries hashes for blocks that are imported
 	BlocksSubscription
-	// LastSubscription keeps track of the last index
+	// LastIndexSubscription keeps track of the last index
```
Comment (palango) [inline eth/filters/filter_system.go:55]: Is this a drive-by change? I don't see it in the upstream PR

**Comment by palango** [comment]
@diwu1989 Thanks for the PR, I added another comment cleaning up some unnecessary parameter.

**Review by carterqw2 — APPROVED** [review]

**Comment by sfc-gh-dwu** [comment]
@palango thanks for the fixup, I'm pretty busy with real life work these days so didn't get around to your feedback. thanks for merging.
